components/esmterrain/storage.cpp

```diff
@@ -385,72 +380,61 @@ namespace ESMTerrain
 
         int rowStart = (origin.x() - cellX) * realTextureSize;
         int colStart = (origin.y() - cellY) * realTextureSize;
-        int rowEnd = rowStart + chunkSize * (realTextureSize-1) + 1;
-        int colEnd = colStart + chunkSize * (realTextureSize-1) + 1;
-
-        // Save the used texture indices so we know the total number of textures
-        // and number of required blend maps
-        std::set<UniqueTextureId> textureIndices;
-        // Due to the way the blending works, the base layer will bleed between texture transitions so we want it to be a black texture
-        // The subsequent passes are added instead of blended, so this gives the correct result
-        textureIndices.insert(std::make_pair(-1,0)); // -1 goes to tx_black_01
-
-        LandCache cache;
-
-        for (int y=colStart; y<colEnd; ++y)
-            for (int x=rowStart; x<rowEnd; ++x)
-            {
-                UniqueTextureId id = getVtexIndexAt(cellX, cellY, x, y, cache);
-                textureIndices.insert(id);
-            }
-
-        // Makes sure the indices are sorted, or rather,
-        // retrieved as sorted. This is important to keep the splatting order
-        // consistent across cells.
-        std::map<UniqueTextureId, int> textureIndicesMap;
-        for (std::set<UniqueTextureId>::iterator it = textureIndices.begin(); it != textureIndices.end(); ++it)
-        {
-            int size = textureIndicesMap.size();
-            textureIndicesMap[*it] = size;
-            layerList.push_back(getLayerInfo(getTextureName(*it)));
-        }
-
-        // size-1 since the base layer doesn't need blending
-        int numBlendmaps = textureIndices.size() - 1;
 
-        // Second iteration - create and fill in the blend maps
         const int blendmapSize = (realTextureSize-1) * chunkSize + 1;
         // We need to upscale the blendmap 2x with nearest neighbor sampling to look like Vanilla
         const int imageScaleFactor = 2;
         const int blendmapImageSize = blendmapSize * imageScaleFactor;
 
-        for (int i=0; i<numBlendmaps; ++i)
-        {
-            osg::ref_ptr<osg::Image> image (new osg::Image);
-            image->allocateImage(blendmapImageSize, blendmapImageSize, 1, GL_ALPHA, GL_UNSIGNED_BYTE);
-            unsigned char* pData = image->data();
+        LandCache cache;
+        std::map<UniqueTextureId, unsigned int> textureIndicesMap;
 
-            for (int y=0; y<blendmapSize; ++y)
+        for (int y=0; y<blendmapSize; ++y)
```

Comment:
Starting to nitpick. Making integer copies (4 bytes) is cheaper on 64-bit architecture than making integer references (8 bytes).